perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -2021,4 +2049,279 @@ public static MapSqlParameterSource getMapSqlParameterSourceToSearchUsersOrMembe
+		sendIdentityRemovedAlert(sess, removedUes, uesAttrs, uesMail,
+				UES_REMOVED_UES_MAIL, DEFAULT_IDENTITY_REMOVED_UES_MAIL,
+				UES_REMOVED_UES_MAIL_SUBJECT, DEFAULT_IDENTITY_REMOVED_UES_MAIL_SUBJECT);
+	}
+
+	/**
+	 * Sends alert to the preferred mail of the user whose identity has bee removed.
+	 *
+	 * @param sess session
+	 * @param removedUes removed userExtSource
+	 * @param uesAttrs attributes of the removed userExtSource
+	 * @param userEmail email, where the message is send
+	 */
+	private static void sendIdentityRemovedAlertToPreferredMail(PerunSession sess, UserExtSource removedUes,
+	                                                            List<Attribute> uesAttrs, String userEmail) {
+		sendIdentityRemovedAlert(sess, removedUes, uesAttrs, userEmail,
+				UES_REMOVED_PREF_MAIL, DEFAULT_IDENTITY_REMOVED_PREF_MAIL,
+				UES_REMOVED_PREF_MAIL_SUBJECT, DEFAULT_IDENTITY_REMOVED_PREF_MAIL_SUBJECT);
+	}
+
+	/**
+	 * Sends alert to the preferred mail of the user of the newly added identity.
+	 *
+	 * @param sess session
+	 * @param newUes new userExtSource
+	 * @param userEmail email, where the message is send
+	 */
+	private static void sendIdentityAddedAlertToPreferredMail(PerunSession sess, UserExtSource newUes, String userEmail) {
+		sendIdentityAddedAlert(sess, newUes, userEmail,
+				UES_ADDED_PREFERRED_MAIL, DEFAULT_IDENTITY_ADDED_PREF_MAIL,
+				UES_ADDED_PREFERRED_MAIL_SUBJECT, DEFAULT_IDENTITY_ADDED_PREF_MAIL_SUBJECT);
+	}
+
+	/**
+	 * Sends alert to the email of the new given userExtSource.
+	 *
+	 * @param sess session
+	 * @param newUes new userExtSource
+	 * @param uesMail email of the newUserExtSource
+	 */
+	private static void sendIdentityAddedAlertToNewIdentityMail(PerunSession sess, UserExtSource newUes, String uesMail) {
+		sendIdentityAddedAlert(sess, newUes, uesMail,
+				UES_ADDED_UES_MAIL, DEFAULT_IDENTITY_ADDED_UES_MAIL,
+				UES_ADDED_UES_MAIL_SUBJECT, DEFAULT_IDENTITY_ADDED_UES_MAIL_SUBJECT);
+	}
+
+	/**
+	 * Sends alert about the given ues being removed.
+	 *
+	 * @param sess session
+	 * @param removedUes the removed ues
+	 * @param uesAttrs attributes of the removed ues
+	 * @param email email, where the alert is send
+	 * @param templateName name of the template for the message
+	 * @param defaultTemplate default template for the message
+	 * @param subjectName name of the subject template
+	 * @param defaultSubject default template for the subject
+	 */
+	private static void sendIdentityRemovedAlert(PerunSession sess, UserExtSource removedUes, List<Attribute> uesAttrs,
+	                                             String email, String templateName, String defaultTemplate,
+	                                             String subjectName, String defaultSubject) {
+		String message = getAlertTemplate(sess, templateName, defaultTemplate);
+		message = message.replaceAll(LOGIN_PLACEHOLDER, removedUes.getLogin());
```

Comment:
I don't think that you need to use `replaceAll()`, just use standard `replace()`. Then you can simplify the constants to plain text form by removing `\\` in them. This apply to all new occurrences.

Reply:
Done.